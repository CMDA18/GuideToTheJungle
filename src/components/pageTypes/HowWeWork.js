// @flow
import React, { Component } from 'react'
import Helmet from 'react-helmet'
import styled from 'styled-components'
import PageTitle from '../PageTitle'
import Intro from '../Intro'
import Grid from '../Grid'
import { howWeWork } from '../../constants/content/index'
import TextAndImage from '../TextAndImage'
import ButtonPrimary from '../../shared/buttons/ButtonPrimary'

type Props = {
  test: string
}

class HowWeWork extends Component<Props> {
  render () {
    return (
      <div className='HowWeWork' data-test={this.props.test}>
        <Helmet
          title='HowWeWork'
        />
        <Grid page='howWeWork'>
          <header>
            <PageTitle {...howWeWork}/>
            <Intro {...howWeWork}/>
          </header>
          <TextAndImage {...howWeWork.textMedia} />
          <ButtonWrapper>
            <ButtonPrimary
              type='internalLink'
              label='Begin module 4'
              route='/ourMethode'
            />
          </ButtonWrapper>
        </Grid>
      </div>
    )
  };
}

export default HowWeWork

const ButtonWrapper = styled.div`
  display: flex;
  margin: 20px auto 0;
  justify-content: center;
`
