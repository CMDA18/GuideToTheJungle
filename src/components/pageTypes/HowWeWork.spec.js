import React from 'react'
import { shallow } from 'enzyme'
import HowWeWork from './HowWeWork'

it('renders without crashing', () => {
  expect(shallow(<HowWeWork />)).toMatchSnapshot()
})
