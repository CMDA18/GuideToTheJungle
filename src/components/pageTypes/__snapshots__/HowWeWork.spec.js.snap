// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HowWeWork />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
          title="HowWeWork"
        />,
        <HowWeWork__Wrapper>
          <Navigation />
          <MainContentGrid>
            <header>
              <PageTitle
                introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                introTitle="One methode"
                pageTitle="0.4 How We Work"
              />
              <Intro
                introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                introTitle="One methode"
                pageTitle="0.4 How We Work"
              />
            </header>
          </MainContentGrid>
        </HowWeWork__Wrapper>,
      ],
      "className": "HowWeWork",
      "data-test": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defer": true,
          "encodeSpecialCharacters": true,
          "title": "HowWeWork",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Navigation />,
            <MainContentGrid>
              <header>
                <PageTitle
                  introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                  introTitle="One methode"
                  pageTitle="0.4 How We Work"
                />
                <Intro
                  introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                  introTitle="One methode"
                  pageTitle="0.4 How We Work"
                />
              </header>
            </MainContentGrid>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <header>
                <PageTitle
                  introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                  introTitle="One methode"
                  pageTitle="0.4 How We Work"
                />
                <Intro
                  introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                  introTitle="One methode"
                  pageTitle="0.4 How We Work"
                />
              </header>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <PageTitle
                    introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                    introTitle="One methode"
                    pageTitle="0.4 How We Work"
                  />,
                  <Intro
                    introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                    introTitle="One methode"
                    pageTitle="0.4 How We Work"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "introText": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi.",
                    "introTitle": "One methode",
                    "pageTitle": "0.4 How We Work",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "introText": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi.",
                    "introTitle": "One methode",
                    "pageTitle": "0.4 How We Work",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "header",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
            title="HowWeWork"
          />,
          <HowWeWork__Wrapper>
            <Navigation />
            <MainContentGrid>
              <header>
                <PageTitle
                  introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                  introTitle="One methode"
                  pageTitle="0.4 How We Work"
                />
                <Intro
                  introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                  introTitle="One methode"
                  pageTitle="0.4 How We Work"
                />
              </header>
            </MainContentGrid>
          </HowWeWork__Wrapper>,
        ],
        "className": "HowWeWork",
        "data-test": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defer": true,
            "encodeSpecialCharacters": true,
            "title": "HowWeWork",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Navigation />,
              <MainContentGrid>
                <header>
                  <PageTitle
                    introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                    introTitle="One methode"
                    pageTitle="0.4 How We Work"
                  />
                  <Intro
                    introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                    introTitle="One methode"
                    pageTitle="0.4 How We Work"
                  />
                </header>
              </MainContentGrid>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <header>
                  <PageTitle
                    introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                    introTitle="One methode"
                    pageTitle="0.4 How We Work"
                  />
                  <Intro
                    introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                    introTitle="One methode"
                    pageTitle="0.4 How We Work"
                  />
                </header>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <PageTitle
                      introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                      introTitle="One methode"
                      pageTitle="0.4 How We Work"
                    />,
                    <Intro
                      introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                      introTitle="One methode"
                      pageTitle="0.4 How We Work"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "introText": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi.",
                      "introTitle": "One methode",
                      "pageTitle": "0.4 How We Work",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "introText": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi.",
                      "introTitle": "One methode",
                      "pageTitle": "0.4 How We Work",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "header",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
